sycl/include/sycl/ext/intel/esimd/detail/memory_intrin.hpp

```diff
@@ -639,7 +656,8 @@ __esimd_gather_masked_scaled2(SurfIndAliasTy surf_ind, uint32_t global_offset,
     char *SlmBase = I->__cm_emu_get_slm_ptr();
     for (int idx = 0; idx < N; ++idx) {
       if (pred[idx]) {
-        Ty *addr = reinterpret_cast<Ty *>(offsets[idx] + SlmBase);
+        RestoredTy *addr =
+            reinterpret_cast<RestoredTy *>(offsets[idx] + SlmBase);
         retv[idx] = *addr;
```

Comment:
```suggestion
        RestoredTy *addr =
            reinterpret_cast<RestoredTy *>(offsets[idx] + SlmBase);
        reinterpret_cast<RestoredTy*>(retv)[idx] = *addr;
```

Reply:
Floating point number tests are failing with this change - including vadd_1d.cpp.

Reply:
Yes, I misinterpreted retv. This is a vector, not a memory pointer. Please disregard/undo this comment. The original version is correct:
```cpp
        RestoredTy *addr =
            reinterpret_cast<RestoredTy *>(offsets[idx] + SlmBase);
        retv[idx] = *addr;
```